vhd/vhd.go

```diff
@@ -62,13 +63,27 @@ type OpenVirtualDiskParameters struct {
 	Version2 OpenVersion2
 }
 
+// The OpenVersion2 structure was mistakingly changed to have the GetInfoOnly and ReadOnly fields be go bools instead
+// of int32's (they're defined as Windows BOOL's so int32 is the correct type). Fix this mistake here by making the right structure we need to pass
+// and not exporting it. We'll translate to the correct type as needed in any of the Open___ calls.
```

Comment:
I'm fine with this PR and I think it does make more sense to use bools for higher-level calls, but I don't think we should refer to this as "fixing the mistake" in the comment. 

Instead we could just say 
```suggestion
// The higher level `OpenVersion2` struct uses bools to refer to `GetInfoOnly` and `ReadOnly` for ease of use. However,
// the internal windows structure uses `BOOLS` aka int32s for these types. `openVersion2` is used for translating 
// `OpenVersion2` fields to the correct windows internal field types on `Open____` calls. 
```

Reply:
I'd worded it like this as a lot of the calls expose a pretty faithful representation of all of the fields for the underlying windows API so this felt out of place with the rest. But we can mask it as us being go friendly 😆

Reply:
It works out well in the end

Reply:
Changed